src/Essentials/src/WebAuthenticator/WebAuthenticatorResult.shared.cs

```diff
@@ -15,9 +15,16 @@ public WebAuthenticatorResult()
 		}
 
 		/// <include file="../../docs/Microsoft.Maui.Essentials/WebAuthenticatorResult.xml" path="//Member[@MemberName='.ctor'][3]/Docs/*" />
-		public WebAuthenticatorResult(Uri uri)
+		public WebAuthenticatorResult(Uri uri) : this(uri, null)
 		{
-			foreach (var kvp in WebUtils.ParseQueryString(uri.ToString()))
+		}
+
+		/// <include file="../../docs/Microsoft.Maui.Essentials/WebAuthenticatorResult.xml" path="//Member[@MemberName='.ctor'][4]/Docs/*" />
+		public WebAuthenticatorResult(Uri uri, IWebAuthenticatorResponseDecoder responseDecoder)
+		{
+			CallbackUri = uri;
+			var properties = responseDecoder == null ? WebUtils.ParseQueryString(uri.ToString()) : responseDecoder.DecodeResponse(uri);
```

Comment:
```suggestion
			var properties = responseDecoder?.DecodeResponse(uri) ?? WebUtils.ParseQueryString(uri.ToString());
```

Maybe we can do this (if my logic is correct) so if the decoder that you have provided does not support the format - for whatever reason - you can just return null and fall back to the default? Or, maybe we should throw?

If we do allow throwing, I wonder if we should have a try/catch around the constructor and pass the exception back into all the TCS?

```cs
try {
tcs.SetResult(new Result(uri, decoder));
} catch (Exception ex) {
tcs.SetException(ex);
}
```

Reply:
Great points - I'll make all the changes here and do the wrapping of the exception at the same time.